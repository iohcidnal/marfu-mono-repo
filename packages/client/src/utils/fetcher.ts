interface FetchConfig {
  url: string;
  method?: 'GET' | 'POST' | 'PUT' | 'DELETE';
  payload?: any;
}

interface FetchResult {
  status: number;
  statusText: string;
  data: any;
}

export default async function fetcher(
  config: FetchConfig = { url: null, method: 'GET', payload: null },
  requestContext = null
): Promise<FetchResult> {
  const option: RequestInit = {
    method: config.method,
    mode: 'cors',
    credentials: 'include',
    headers: {
      'Content-Type': 'application/json'
    },
    ...(config.payload && { body: JSON.stringify(config.payload) })
  };

  if (requestContext) {
    // Use [] notation to make TS happy.
    option.headers['cookie'] = requestContext.req.headers.cookie;
  }

  console.log('option :>> ', option);
  const response: Response = await fetch(config.url, option);
  console.log('response :>> ', response);

  const data = await response.json();

  return {
    status: response.status,
    statusText: response.statusText,
    data
  };
}
