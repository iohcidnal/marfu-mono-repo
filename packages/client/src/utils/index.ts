export { default as fetcher } from './fetcher';
export type { IFetchResult } from './fetcher';
export { default as renderWithProviders } from './test-utils';
