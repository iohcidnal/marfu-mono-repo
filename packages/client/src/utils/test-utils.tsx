import * as React from 'react';
import { render } from '@testing-library/react';
import { ChakraProvider } from '@chakra-ui/react';
import { QueryClientProvider, QueryClient } from 'react-query';

const queryClient = new QueryClient();

export default function renderWithProviders(children: React.ReactNode) {
  const Component = () => (
    <QueryClientProvider client={queryClient}>
      <ChakraProvider>{children}</ChakraProvider>
    </QueryClientProvider>
  );

  return () => {
    const rendered = render(<Component />);
    return {
      ...rendered
    };
  };
}
