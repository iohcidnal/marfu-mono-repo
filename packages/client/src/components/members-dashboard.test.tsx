import { act } from '@testing-library/react';
import { renderWithProviders } from '../utils';
import MembersDashboard from './members-dashboard';

const render = renderWithProviders(<MembersDashboard members={[]} />);

describe('SignIn', () => {
  it('should render properly', async () => {
    let rendered;
    await act(async () => {
      rendered = render();
    });
    expect(rendered.asFragment()).toMatchSnapshot();
  });

  it.todo('cover other lines');
});
