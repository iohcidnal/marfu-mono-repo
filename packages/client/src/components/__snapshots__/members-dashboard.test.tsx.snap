// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn should render properly 1`] = `<DocumentFragment />`;
